onboard/catkin_ws/src/motion_planning/scripts/combination_tasks.py

```diff
@@ -40,17 +40,17 @@ def _on_task_run(self):
             self.finish()
 
 
-class MasterSlaveTask(Task):
+class ParentChildTask(Task):
```

Comment:
Hmm this is a fair point. Samuel and I discussed both options and thought leader-follower was less intuitive because it's not obvious that the leader has complete control over the follower. I think I like leader-follower better given the point about spawning. @rabinowitz thoughts?

Reply:
My issues with leader-follower are:
1. It doesn't establish enough control of the leader over the follower (a concern conveyed [here](https://news.ycombinator.com/item?id=7801646) if you ignore the politically-charged comments).
1. In distributed systems, there are [leader elections](https://en.wikipedia.org/wiki/Leader_election), so followers can become elected leader and then all other followers switch leaders. That doesn't happen here.
1. In the case of our task planning, in which things often happen sequentially, my first (and I'm assuming others would do this as well) thought when I hear "Leader-Follower" is that the follower does the same thing as the leader _after_ the leader, like baby ducks following their momma.

But I also know that parent-child is used in the Unix [fork](https://en.wikipedia.org/wiki/Fork_(system_call)) call, in which child processes are created as copies of the parent and become orphaned if the parent dies first (see [here](https://en.wikipedia.org/wiki/Parent_process)), which isn't how ours works.

To me, there doesn't seem to be a good pick. Master-slave still is the most unambiguous descriptor. So, since we're choosing between two worse options, and simultaneously *kinda* helping define the standard that is only recently changing, we can go with whatever we want. I'm still partial to parent-child here.

Reply:
Eric and I discussed, since both aren't perfect replacements, but Leader-Follower has become the standard drop-in replacement for Master-Slave and probably won't cause _that_ much confusion, we're going to go with that.